acvm_backend_barretenberg/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+#![warn(unused_crate_dependencies, unused_extern_crates)]
+#![warn(unreachable_pub)]
+
+// `acvm-backend-barretenberg` can either interact with the Barretenberg backend through a static library
+// or through an embedded wasm binary. It does not make sense to include both of these backends at the same time.
+// We then throw a compilation error if both flags are set.
+// TODO: handle JS target.
+#[cfg(all(feature = "native", feature = "wasm"))]
+compile_error!("feature \"native\"  and feature \"wasm\" cannot be enabled at the same time");
+
+mod acvm_interop;
+mod blake2s;
+mod composer;
+#[cfg(all(feature = "native", test))]
```

Comment:
This is in native and wasm, right? It just doesn't work in JS right now

Reply:
The first of these tests require native features to run and both will quickly be invalidated as we change out how we handle the CRS. I'm inclined to leave it as is and we then fix it properly as we're writing that new code.

Reply:
Ah, this is only for the tests, ignore me then